From a80462858705f4b2613261715a2335a81f2cb9f0 Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Sun, 12 Mar 2023 21:02:42 +0000
Subject: [PATCH] treat DEFENV_VERILATOR_ROOT as c_str for conda relocatability

---
 src/V3Options.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/V3Options.cpp b/src/V3Options.cpp
index faec6ba50..375e86ab9 100644
--- a/src/V3Options.cpp
+++ b/src/V3Options.cpp
@@ -725,7 +725,7 @@ string V3Options::getenvSYSTEMC_LIBDIR() {
 string V3Options::getenvVERILATOR_ROOT() {
     string var = V3Os::getenvStr("VERILATOR_ROOT", "");
     if (var == "" && string(DEFENV_VERILATOR_ROOT) != "") {
-        var = DEFENV_VERILATOR_ROOT;
+        var = string(DEFENV_VERILATOR_ROOT).c_str();
         V3Os::setenvStr("VERILATOR_ROOT", var, "Hardcoded at build time");
     }
     if (var == "") v3fatal("$VERILATOR_ROOT needs to be in environment\n");
-- 
2.39.1

